### PR TITLE
Improved Error Handling in `remap_file` Function and Added `.DS_Store` to `.gitignore

### PR DESCRIPTION
**Pull request by mendax0110**
### Description:

This pull request introduces two significant changes:

**Improved Error Handling in `remap_file` Function:**
- The error handling in the `remap_file` function has been simplified for improved clarity and user-friendliness. The original code included a combination of `assert` and `exit`, which could be perplexing during runtime.
  
**Changes Made:**
- Replaced the original `assert` and `exit` with a more straightforward `if-else` block for handling errors.
- Utilized `sys.exit` for clear error messages and proper program termination when an error occurs.

**Why I Did It:**
- Improved Clarity: The original code's `assert` primarily served debugging purposes and might not behave as expected during runtime. The new code offers clarity and predictability.
- User-Friendly Errors: The updated code provides users with easily understandable error messages.

**Added `.DS_Store` to `.gitignore`:**
- `.DS_Store` files are commonly generated by macOS Finder to store folder-specific metadata. They do not need to be tracked in version control.
  
**Changes Made:**
- Added `.DS_Store` to the `.gitignore` file.

**Why I Did It:**
- To prevent `.DS_Store` files from being tracked in the Git repository, ensuring a cleaner repository and avoiding unintended commits of macOS-specific files.

### Improvements:

- **Better Error Handling:** The new code in the `remap_file` function ensures the program exits gracefully with informative error messages when errors occur.
- **Enhanced Code Clarity:** The code now communicates its intentions more clearly with `sys.exit` for handling expected runtime errors.
- **Cleaner Git Repository:** The addition of `.DS_Store` to `.gitignore` prevents the tracking of macOS-specific files in version control.

### CODE REVIEW

File: scripts/fix_cc_deps.py [hunk 1]
```diff
@@ -94,9 +95,11 @@ def remap_file(label: str) -> str:
     repo, _, path = label.partition("//")
     if not repo:
         return path.replace(":", "/")
-    assert repo in EXTERNAL_REPOS, repo
-    return EXTERNAL_REPOS[repo].remap(path)
-    exit(f"Don't know how to remap label '{label}'")
```
Comment (jonmeow) [inline scripts/fix_cc_deps.py]: Right now, it looks like this exit is unreachable. The assert is just there as a safety check to expose the `repo` value.

Is there an issue that you ran into that made you want to expose the currently-unreachable exit?

Comment (mendax0110) [reply scripts/fix_cc_deps.py]: I haven't encountered any issues so far, but I just wanted to highlight that the exit statement in the code is currently not reachable.

Comment (jonmeow) [reply scripts/fix_cc_deps.py]: Okay, I think it'd make sense to just remove the unreachable code.

Comment (mendax0110) [reply scripts/fix_cc_deps.py]: Got it, i just removed the unreachable code. :)

File: scripts/fix_cc_deps.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 """
 
+import sys
```
Comment (jonmeow) [inline scripts/fix_cc_deps.py]:
```suggestion
```

Looks like this got left in by accident.

**Review by jonmeow — APPROVED** [review]
Thanks, looks good! Just one last small thing.

**Pull request merged**